packages/world-modules/test/SystemSwitch.t.sol

```diff
@@ -196,6 +204,12 @@ contract SystemSwitchTest is Test, GasReporter {
     assertEq(abi.decode(returnData, (string)), "hello");
   }
 
+  function testCallNonRootFromRootIndex() public {
+    vm.prank(caller);
+    bytes memory returnData = _executeFromRootSystemA(systemBId, abi.encodeCall(EchoSystem.index, ()));
+    assertEq(abi.decode(returnData, (uint256)), INDEX);
+  }
```

Comment:
I couldn't think of a good name for this variable, I agree it's confusing.

The goal of this test is to ensure `EchoSystem.index()` is executed with `call` by the root system. Reading the `index` storage variable is a quick way to check this - if the function is wrongly executed with `delegatecall`, it will attempt to read the Worlds storage (which doesn't have an `index` variable), and return the wrong value.

Reply:
A more MUD-native way of doing this would be to attempt to write to a table that the non-root system doesn't have access to.

Reply:
what about something really obvious like `exampleContractState`? and a comment that "systems shouldn't have state, we're just testing the behavior of calling systems in different contexts"

Reply:
I've just flipped it around so the system attempts to read a table in it's own storage. This should return nothing and fail if `call` is used on a non-root system (because they are not Stores).